components/Buttons/src/MDCFloatingButton+Animation.m

```diff
@@ -89,110 +86,190 @@ + (float)fab_dragCoefficient {
 }
 #endif
 
++ (CFTimeInterval)fab_animationDurationFromWallTimeDuration:(CFTimeInterval)duration
+                                                   forLayer:(CALayer *)layer {
+  CFTimeInterval finalDuration = duration;
+
+#if TARGET_IPHONE_SIMULATOR
+  finalDuration *= [self fab_dragCoefficient];
+#endif
```

Comment:
A follow-up change may help simplify the utility of this dragCoefficient API if you move to the pattern we employ in Material Motion: https://github.com/material-motion/motion-animator-objc/blob/develop/src/MDMMotionAnimator.m#L21

Notably you won't need preprocessor macros throughout the code.

Reply:
Is that approach favored over the one in [MDCFlexibleHeaderView](https://github.com/material-components/material-components-ios/blob/develop/components/FlexibleHeader/src/MDCFlexibleHeaderView.m#L602)?